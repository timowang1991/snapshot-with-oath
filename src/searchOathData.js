const oathDb = {
    Arvin: {
        image: require('./assets/arvin.png'),
        oath: ['Workout every night', 'I hate reflection']
    },
    Cloe: {
        image: require('./assets/cloe.jpg'),
        oath: ['Shut Timo up']
    },
    John: {
        image: require('./assets/john.png'),
        oath: ['Why so serious?']
    },
    Kate: {
        oath: ['Work from remote']
    },
    'KP': {
        image: require('./assets/kp.jpg'),
        oath: ['Banish all bast**ds']
    },
    Sonys: {
        image: require('./assets/sonys.jpg'),
        oath: ['Youcard is MINE!']
    },
    Timo: {
        image: require('./assets/timo.jpg'),
        oath: ['Listen to TA', 'Stop global warming']
    },
    'Tim Armstrong': {
        image: require('./assets/tim-armstrong.jpg'),
        oath: ['Give up!', 'Take private jets']
    },
    'Tsai Ing-wen': {
        image: require('./assets/tsai-ing-wen.jpg'),
        oath: ['No more blackouts']
    }
};

export default (query) => {
    if (!query || typeof query !== 'string') {
        return Object.keys(oathDb).map((name) => ({ name, info: oathDb[name] }));
    }
    const lowerCaseQuery = query.toLowerCase();
    const names = Object.keys(oathDb).filter((personName) =>
        personName.toLowerCase().indexOf(lowerCaseQuery) === 0
    );
    return names.map((name) => ({ name, info: oathDb[name] }));
}