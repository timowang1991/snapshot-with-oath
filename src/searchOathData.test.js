import searchOathData from './searchOathData';

it('should return name starting with T', () => {
    const oath = searchOathData('T');
    
    expect(oath).toMatchSnapshot();
});

it('should return name with sequences of Timo', () => {
    ['T', 'Ti', 'Timo'].forEach((query) => {
        const oath = searchOathData(query);

        expect(oath).toMatchSnapshot();
    });
});
