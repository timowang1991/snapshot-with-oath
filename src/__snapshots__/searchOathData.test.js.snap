// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return name starting with T 1`] = `
Array [
  Object {
    "info": Object {
      "image": "timo.jpg",
      "oath": Array [
        "Listen to TA",
        "Stop global warming",
      ],
    },
    "name": "Timo",
  },
  Object {
    "info": Object {
      "image": "tim-armstrong.jpg",
      "oath": Array [
        "Give up!",
        "Take private jets",
      ],
    },
    "name": "Tim Armstrong",
  },
  Object {
    "info": Object {
      "image": "tsai-ing-wen.jpg",
      "oath": Array [
        "No more blackouts",
      ],
    },
    "name": "Tsai Ing-wen",
  },
]
`;

exports[`should return name with sequences of Timo 1`] = `
Array [
  Object {
    "info": Object {
      "image": "timo.jpg",
      "oath": Array [
        "Listen to TA",
        "Stop global warming",
      ],
    },
    "name": "Timo",
  },
  Object {
    "info": Object {
      "image": "tim-armstrong.jpg",
      "oath": Array [
        "Give up!",
        "Take private jets",
      ],
    },
    "name": "Tim Armstrong",
  },
  Object {
    "info": Object {
      "image": "tsai-ing-wen.jpg",
      "oath": Array [
        "No more blackouts",
      ],
    },
    "name": "Tsai Ing-wen",
  },
]
`;

exports[`should return name with sequences of Timo 2`] = `
Array [
  Object {
    "info": Object {
      "image": "timo.jpg",
      "oath": Array [
        "Listen to TA",
        "Stop global warming",
      ],
    },
    "name": "Timo",
  },
  Object {
    "info": Object {
      "image": "tim-armstrong.jpg",
      "oath": Array [
        "Give up!",
        "Take private jets",
      ],
    },
    "name": "Tim Armstrong",
  },
]
`;

exports[`should return name with sequences of Timo 3`] = `
Array [
  Object {
    "info": Object {
      "image": "timo.jpg",
      "oath": Array [
        "Listen to TA",
        "Stop global warming",
      ],
    },
    "name": "Timo",
  },
]
`;
